components/MediList/index.js

```diff
@@ -63,9 +82,24 @@ const Form = styled.form`
 
 const DosageInputContainer = styled.div`
   display: flex;
-  align-items: center; /* Vertically center the input and button */
+  align-items: center; 
 `;
 
+const SearchContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-end;
+  margin-bottom: 10px;
+`;
+
+const SearchButton = styled.button`
+  margin-left: 10px;
+`;
+
+const MedicationListContainer = styled.div`
+  max-height: 300px; 
+  overflow: auto; 
+`;
 
 const MedikamentWrapper = styled.div`
```

Comment:
The MedicationWrapper is your single medication item. It could be a "li" to make your app more semantic.